Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -49,6 +49,9 @@ class NodejsProjectNode : CommonProjectNode, VsWebSite.VSWebSite, INodePackageMo
         internal readonly RequireCompletionCache _requireCompletionCache = new RequireCompletionCache();
         private string _intermediateOutputPath;
         private readonly Dictionary<NodejsProjectImageName, int> _imageIndexFromNameDictionary = new Dictionary<NodejsProjectImageName, int>();
+        private bool _projectHasTypeScriptFiles = false;
```

Comment:
`EnableTypescript` can only be called on a UI thread, which I realized after extracting out duplicate code for that property.